src/amo/components/OverallRating.js

```diff
@@ -1,37 +1,69 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
+import { setUserRating } from 'amo/actions/ratingActions';
+import { callApi } from 'core/api';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
 
 import 'amo/css/OverallRating.scss';
 
 
 export class OverallRatingBase extends React.Component {
   static propTypes = {
-    i18n: PropTypes.object,
-    addonName: PropTypes.string,
+    addonName: PropTypes.string.isRequired,
+    addonID: PropTypes.number.isRequired,
+    authToken: PropTypes.string,
+    version: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    createRating: PropTypes.func.isRequired,
+  }
+
+  onChange = ({ currentTarget: form }) => {
+    const input = form.querySelector('input:checked');
+    if (!input) {
+      // I don't think any user input would cause a form change like
+      // this. I'm not sure what would cause this at all but it's probably
+      // better not to throw an exception.
+      log.error('The form changed but no input was selected');
```

Comment:
For a list of choices a radio button seems like the best fit. However the submission `onchange` seems a little bit problematic. A radio button will submit per change, that includes when using the keyboard to select the choice.  For example see http://jsbin.com/dezono/edit?html,js,console,output and use the keyboard to focus the radio and then use the arrow keys to select an option. 

A set of radio buttons really only makes sense if we have a button to submit the form and it seems we don't have that at the moment, so to continue with them I think we'd have to add a submit button and not make a submission of data `onchange`.

The alternative of using buttons that Mark suggested is interesting, but I think we'd need to consider any accessibility issues around that approach. It does allow picking the choice before submission but the bigger issue I see with this is, it's not really an expected UI for choices. 

Adding a submit button but keeping the radio buttons is probably the simplest approach for now. I would suggest doing that since we know that it would be accessible and then investigate alternatives separately, where we can get UX and A11y people involved.